### PR TITLE
Rhel8

### PR DESCRIPTION
**Pull request by robertc99**
Second attempt at rhel8 support.
Leave out the default parameter fixes.
API unchanged except for added authselect_profile parameter with sensible default so it won't break existing config.

### CODE REVIEW

**Comment by ghoneycutt** [comment]
You are pushing a ton of commits up and I'm guessing waiting on travis for feedback. Can I help you with setting up the testing locally so that development is easier?

File: REFERENCE.md [hunk 1]
```diff
@@ -189,3 +189,11 @@ Default value: [
     '--disablesssdauth',
   ]
 
+##### `authselect_profile`
```
Comment (ghoneycutt) [inline REFERENCE.md]: Don't modify this file. it is auto generated during the deployment process.

File: manifests/init.pp [hunk 2]
```diff
@@ -46,6 +46,8 @@
 # @param ensure_absent_flags Array of flags to use with authconfig when service
 #   is disabled.
 #
+# @param authselect_profile  Name of authselect profile to use
```
Comment (ghoneycutt) [inline manifests/init.pp]: What is this for? Is it something new in RHEL8? Could you link to some of the RHEL8 documentation that explains it.

File: manifests/init.pp [hunk 3]
```diff
@@ -85,15 +87,19 @@
...
   if ($::facts['os']['family'] == 'RedHat') {
     if ($::facts['os']['name'] == 'Amazon') and !($::facts['os']['release']['major'] in ['2']) {
       warning("osname Amazon's os.release.major is <${::facts['os']['release']['major']}> and must be 2.")
     }
-    if !($::facts['os']['name'] == 'Amazon') and !($::facts['os']['release']['major'] in ['6', '7']) {
-      warning("osfamily RedHat's os.release.major is <${::facts['os']['release']['major']}> and must be 6 or 7.")
+    if !($::facts['os']['name'] == 'RedHat') and !($::facts['os']['release']['major'] in ['6', '7', '8']) {
+      warning("osname RedHat's os.release.major is <${::facts['os']['release']['major']}> and must be 6, 7 or 8.")
+    }
+    if !($::facts['os']['name'] == 'Fedora') and !($::facts['os']['release']['major'] >= '28') {
```
Comment (ghoneycutt) [inline manifests/init.pp]: This would add support for Fedora 28 though the pull request is for RHEL8. This should be in a separate pull request. By adding it here, it confuses the two and makes both of them harder to merge. Also F28 is end of life, so we wouldn't merge it.

https://fedoraproject.org/wiki/End_of_life

File: manifests/init.pp (hunk 2 above)
Comment (robertc99) [reply manifests/init.pp]: No, this is in Fedora also.
Authselect allows you to change between authentication "profiles"
Fedora/RHEL8 ships with a handful of predefined profiles including an sssd profile.
But you can also define your own.
We use a locally defined profile with does sssd but adds MFA as well.

File: manifests/init.pp (hunk 3 above)
Comment (robertc99) [reply manifests/init.pp]: Dropped, I was trying to make the code match the doco. But I put fedora 28 instead of fedora 29.

File: REFERENCE.md (hunk 1 above)
Comment (robertc99) [reply REFERENCE.md]: dropped

File: spec/classes/init_spec.rb [hunk 4]
```diff
@@ -569,7 +598,7 @@
           v[:facts_hash]
         end
 
-        if v[:facts_hash][:os]['name'] == 'RedHat'
+        if v[:facts_hash][:os]['name'] == 'RedHat' and  v[:facts_hash][:os]['release']['major'] <= '7'
```
Comment (ghoneycutt) [inline spec/classes/init_spec.rb]: You used `< 8` previously. Should stick with that for consistency.

**Comment by ghoneycutt** [comment]
Checkout out the `.travis.yml` file. This will need a `spec/acceptance/nodesets` Checkout https://github.com/ghoneycutt/puppet-module-pam/pull/213/files#diff-872af31a2532e02c860620601da2b831

File: spec/classes/init_spec.rb (hunk 4 above)
Comment (robertc99) [reply spec/classes/init_spec.rb]: done

**Comment by robertc99** [comment]
I don't really understand the nodeset/travis stuff.
Is that what you wanted?

File: .travis.yml [hunk 5]
```diff
@@ -41,6 +41,18 @@ matrix:
     bundler_args:
     script: bundle exec rake beaker
   - rvm: 2.4.4
+    sudo: required
+    services: docker
+    env: BEAKER_set="centos-8" BEAKER_PUPPET_COLLECTION=puppet5
+    bundler_args:
+    script: bundle exec rake beaker
+  - rvm: 2.5.1
```
Comment (ghoneycutt) [inline .travis.yml]: 2.5.3

File: .travis.yml [hunk 6]
```diff
@@ -112,7 +124,12 @@ matrix:
     env: BEAKER_set="ubuntu-1804" BEAKER_PUPPET_COLLECTION=puppet6
     bundler_args:
     script: bundle exec rake beaker
-
+  allow_failures:
+  # waiting for centos 8 release
```
Comment (ghoneycutt) [inline .travis.yml]: These should work if we're going to support it.

Comment (robertc99) [reply .travis.yml]: except centos8 hasnt been released yet.
I was just copying your example.
Do you want me to try and change it to rhel8?

**Comment by dacron** [comment]
Be real nice to get this merge and a new release cut :)

File: .travis.yml (hunk 6 above)
Comment (ghoneycutt) [reply .travis.yml]: Yes, please. We want to ensure these tests are passing.

File: metadata.json [hunk 7]
```diff
@@ -15,7 +15,7 @@
     },
     {
       "operatingsystem": "RedHat",
-      "operatingsystemrelease": ["5","6","7","26", "27"]
+	"operatingsystemrelease": ["5","6","7","8","26", "27"]
```
Comment (ghoneycutt) [inline metadata.json]: this indentation is two spaces too long

**Comment by ghoneycutt** [comment]
Hi @robertc99 

Think we just have a couple minor changes and I'll get these merged. If you can squash your commits that would also be helpful. If not, I can handle that.

File: metadata.json (hunk 7 above)
Comment (robertc99) [reply metadata.json]: fixed. The 26,27 is dubious. Those are fedora version numbers. And not even ones
your really supporting.
